Source/WebKit/GPUProcess/graphics/WebGPU/RemoteDevice.cpp

```diff
@@ -152,24 +153,36 @@ void RemoteDevice::createSampler(const WebGPU::SamplerDescriptor& descriptor, We
     m_objectHeap.addObject(identifier, remoteSampler);
 }
 
+static void populateConvertedDescriptor(auto mediaIdentifier, auto& convertedDescriptor, auto& performWithMediaPlayerOnMainThread)
+{
+    using MediaPlayerOrVideoFrameResult = std::variant<WebCore::MediaPlayerIdentifier, WebKit::RemoteVideoFrameReference>;
+    MediaPlayerOrVideoFrameResult result = WTF::switchOn(mediaIdentifier, [] (PAL::WebGPU::HTMLVideoElementIdentifier i) -> MediaPlayerOrVideoFrameResult {
+        return WebCore::MediaPlayerIdentifier(i.value);
+    }, [] (PAL::WebGPU::WebCodecsVideoFrameIdentifier i) -> MediaPlayerOrVideoFrameResult {
+#if PLATFORM(COCOA)
+        RemoteVideoFrameReference r;
+        memcpy(reinterpret_cast<void*>(&r), &i.value[0], i.value.size());
```

Comment:
can we have a strong release assert that i.value.size() is the same as sizeof(RemoteVideoFrameReference).

Is there no other way to create a RemoteVideoFrameReference? can't we add a constructor instead?

Reply:
I think I can use identifier and version instead to create the RemoteVideoFrameReference, I will try that